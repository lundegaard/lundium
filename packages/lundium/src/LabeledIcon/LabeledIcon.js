import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { cx } from 'ramda-extension';

import Text from '../Text';
import Icon from '../Icon';
import stringOrNumberPropType from '../utils/types/stringOrNumberPropType';

const LabeledIcon = ({
	children,
	iconBeforeText,
	iconType,
	iconColor,
	iconSize,
	iconClassName,
	iconProps: rawIconProps = {},
}) => {
	const iconProps = {
		ml: iconBeforeText ? 2 : 3,
		mr: iconBeforeText ? 3 : 2,
		...rawIconProps,
	};

	const icon = (
		<Icon
			type={iconType}
			color={iconColor}
			size={iconSize}
			className={cx(iconClassName, iconProps.className)}
			{...iconProps}
		/>
	);
	return (
		<Fragment>
			{iconBeforeText && icon}

			<Text as="span" align={iconBeforeText ? 'left' : 'right'}>
				{children}
			</Text>

			{!iconBeforeText && icon}
		</Fragment>
	);
};

LabeledIcon.propTypes = {
	/** Children to be rendered in the main container. */
	children: PropTypes.node,
	iconBeforeText: PropTypes.bool,
	iconClassName: PropTypes.string,
	iconColor: PropTypes.string,
	iconProps: PropTypes.object,
	iconSize: stringOrNumberPropType,
	iconType: PropTypes.string.isRequired,
};

export default LabeledIcon;
