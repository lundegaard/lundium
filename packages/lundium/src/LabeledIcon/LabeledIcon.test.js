import React from 'react';
import { shallow } from 'enzyme';

import LabeledIcon from './LabeledIcon';

describe('<LabeledIcon />', () => {
	const requiredProps = { iconType: 'add' };

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<LabeledIcon {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
