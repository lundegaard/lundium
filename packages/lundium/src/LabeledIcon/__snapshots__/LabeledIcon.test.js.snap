// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LabeledIcon /> matches snapshot when shallowly rendered 1`] = `
<Fragment>
  <forwardRef(Text)
    align="right"
    as="span"
  />
  <Icon
    className=""
    ml={3}
    mr={2}
    type="add"
  />
</Fragment>
`;
